OmniKit/MessageTransport/MessageBlocks/PodInfoFaultEvent.swift

```diff
@@ -47,64 +47,58 @@ public struct StatusError : MessageBlock {
         case immediateBolusInProgressDuringError = 1
         // TODO: bb: internal boolean variable initialized to Tab5[$D] != 0
     }
-    
-    public let requestedType: GetStatusCommand.StatusType
-    public let length: UInt8
-    public let blockType: MessageBlockType = .statusError
-    public let deliveryInProgressType: DeliveryInProgressType
-    public let reservoirStatus: StatusResponse.ReservoirStatus  // Reused from StatusResponse
-    public let insulinNotDelivered: Double
-    public let podMessageCounter: UInt8
-    public let unknownPageCode: Double
-    public let originalLoggedFaultEvent: UInt8
+
+    public let length                   : UInt8
+    public var podInfoType              : PodInfoResponseSubType = .faultEvents
+    public let reservoirStatus          : StatusResponse.ReservoirStatus
+    public let deliveryInProgressType   : DeliveryInProgressType
+    public let insulinNotDelivered      : Double
+    public let podMessageCounter        : UInt8
+    public let unknownPageCode          : Double
+    public let originalLoggedFaultEvent : PodInfoResponseSubType.FaultEventType
     public let faultEventTimeSinceActivation: Double
-    public let insulinRemaining: Double
-    public let timeActive: TimeInterval
-    public let secondaryLoggedFaultEvent: UInt8
-    public let logEventError: Bool
-    public let infoLoggedFaultEvent: InfoLoggedFaultEventType
+    public let insulinRemaining         : Double
+    public let timeActive               : TimeInterval
+    public let secondaryLoggedFaultEvent: PodInfoResponseSubType.FaultEventType
+    public let logEventError            : Bool
+    public let infoLoggedFaultEvent     : InfoLoggedFaultEventType
     public let reservoirStatusAtFirstLoggedFaultEvent: StatusResponse.ReservoirStatus
-    public let recieverLowGain: UInt8
-    public let radioRSSI: UInt8
+    public let recieverLowGain          : UInt8
+    public let radioRSSI                : UInt8
     public let reservoirStatusAtFirstLoggedFaultEventCheck: StatusResponse.ReservoirStatus
-
-    public let data: Data
+    
+    public let data                     : Data
     
     public init(encodedData: Data) throws {
         
-        if encodedData.count < Int(13) {
+        if encodedData.count < Int(16) {
             throw MessageBlockError.notEnoughData
         }
         
         self.length = encodedData[1]
```

Comment:
Because this object is now being constructed with `podInfo = try podInfoResponseSubType.podInfoType.init(encodedData: encodedData.subdata(in: 2..<len-2))`, all the array accesses should be shifted minus 2.

Reply:
Fixed, as well as the tests for these types.